### DISCUSSION TITLE
Understanding Pointers

### DISCUSSION THREAD

**Discussion by proj-sashido** [netreconlab/Parse-Swift#109] [Q&A] [answered]
I have a scenario analogous to the following setup:

Consider a `User` struct:
```swift 
public struct User: ParseUser {
    /**
      All the standard variables.
    */
    public var nickname: String?
    public var metadata: Pointer<Metadata>?
}
```

and a `Metadata` struct, as so:
```swift 
public struct Metadata: ParseObject {
    /**
      All the standard variables.
    */
    public var user: Pointer<User>?
    public var photo: ParseFile?
    public var parentName: String?
}
```
i.e. I want the ability to traverse from a User object to their Metadata object (e.g. to set and get the user’s photo and parent name), and the ability to fetch a Metadata object, and get to that user’s nickname.
I can’t remove the Pointer in both structs, because I would then have a stored property that recursively contains itself.
So I keep them both as Pointers to the other class.

However, if I have a query, say:
```swift
    let query = Metadata.query ( "parentName" == "Dave" )
        .include("user")
    let result = try? await query.first()
```

I notice that I won’t have access to the User object through `result`, I’d have to `result.user?.fetch()` in order to access the User object.
Is there any way for me to also fetch the User object when I use `.include("user")`? Am I doing things sub-optimally?
This question isn’t specific to the User table, but for any two structs that might have pointers to the other.

**Comment by cbaker6** [comment, marked as answer]
> I can’t remove the Pointer in both structs, because I would then have a stored property that recursively contains itself.

The compiler providing this type of error is a suggestion towards a design improvement in your schema (you are basically designing in a way that can create cycles). If you want to keep your current schema setup then the extra fetch you mentioned works as a solution and `include` won't help because if you create a cycle you could end up making your server go in an infinite recursion loop. You can also look at [relations](https://github.com/netreconlab/Parse-Swift/blob/main/ParseSwift.playground/Pages/12%20-%20Roles%20and%20Relations.xcplaygroundpage/Contents.swift), but you will most likely still need the extra fetch. The [pointer](https://github.com/netreconlab/Parse-Swift/blob/main/ParseSwift.playground/Pages/8%20-%20Pointers.xcplaygroundpage/Contents.swift) in Playgrounds demonstrates the solution you posted as well as others.

You can also look at join tables:
- https://stackoverflow.com/q/61391457
